rplugin/python3/deoplete/sources/deoplete_jedi.py

```diff
@@ -33,34 +34,46 @@ def on_init(self, context):
         vars = context['vars']
 
         self.statement_length = vars.get(
-            'deoplete#sources#jedi#statement_length', 0
-        )
+            'deoplete#sources#jedi#statement_length', 0)
         self.use_short_types = vars.get(
-            'deoplete#sources#jedi#short_types', False
-        )
+            'deoplete#sources#jedi#short_types', False)
         self.show_docstring = vars.get(
-            'deoplete#sources#jedi#show_docstring', False
-        )
+            'deoplete#sources#jedi#show_docstring', False)
+        self.debug_server = vars.get(
+            'deoplete#sources#jedi#debug_server', False)
         # Only one worker is really needed since deoplete-jedi has a pretty
         # aggressive cache.
         # Two workers may be needed if working with very large source files.
         self.worker_threads = vars.get(
-            'deoplete#sources#jedi#worker_threads', 1
-        )
+            'deoplete#sources#jedi#worker_threads', 1)
         # Hard coded python interpreter location
         self.python_path = vars.get(
-            'deoplete#sources#jedi#python_path', ''
-        )
+            'deoplete#sources#jedi#python_path', '')
 
         self.workers_started = False
         self.boilerplate = []  # Completions that are included in all results
 
+        log_file = ''
+        root_log = logging.getLogger('deoplete')
+        if not self.debug_enabled:
+            child_log = root_log.getChild('jedi')
+            child_log.propagate = False
+
+        if self.debug_enabled and self.debug_server:
+            if isinstance(self.debug_server, str):
+                log_file = self.debug_server
+            else:
+                for handler in root_log.handlers:
+                    if isinstance(handler, logging.FileHandler):
+                        log_file = getattr(handler, 'baseFilename', None)
+                        break
```

Comment:
> Still not following about "Only consider the "deoplete" root logger." 

I've thought that it might be possible to _only_ look at the deoplete root handler right away.